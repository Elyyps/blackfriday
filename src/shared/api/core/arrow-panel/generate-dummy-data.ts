import { IArrowPanelProps } from "@app/core/arrow-panel";
import party from "@assets/icons/party.svg";
import iears from "@assets/dummy-images/product-thumbnails/iears.png";
import imac from "@assets/dummy-images/product-thumbnails/imac.png";
import iphone from "@assets/dummy-images/product-thumbnails/iphone.png";
import loader from "@assets/dummy-images/product-thumbnails/loader.png";
import macbook from "@assets/dummy-images/product-thumbnails/macbook.png";
import mouse from "@assets/dummy-images/product-thumbnails/mouse.png";

export const generateArrowPanelData = (): IArrowPanelProps[] => {
  const data: IArrowPanelProps[] = [
    {
      image: iears,
      link: { url: "/stores-single", title: "Apple Airpods" }
    },
    {
      image: imac,
      link: { url: "/stores-single", title: "Apple iPhones" }
    },
    {
      image: iphone,
      link: { url: "/stores-single", title: "Apple MacBooks" }
    },
    {
      image: loader,
      link: { url: "/stores-single", title: "Apple iMacs" }
    },
    {
      image: macbook,
      link: { url: "/stores-single", title: "Categorie item met een lange naam" }
    },
    {
      image: mouse,
      link: { url: "/stores-single", title: "Categorie item met een nog langere naam" }
    },
    {
      link: { url: "/stores-single", title: "Categorie item zonder afbeelding" }
    },
    {
      icon: party,
      link: { url: "/stores-single", title: "Categorie met icoon" }
    }
  ];

  return data;
};
