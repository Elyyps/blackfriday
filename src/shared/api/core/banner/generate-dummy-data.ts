import { Banner } from "./banner";
import CoolBlue from "@assets/cool-blue.svg";

export const generateDummyBannerComponentData = (): Banner => ({
  title: "Black Friday bij Coolblue",
  buttonTitle: "Naar deals",
  buttonLink: "/deals-overview",
  image: {
    src: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/ad.png",
    alt: "Ad picture"
  },
  imageLink: "https://coolblue.nl",
  label: "22 t/m 26 nov.",
  logo: { src: CoolBlue, alt: "CoolBlue logo" },
  text: "Een selectie van bizarre aanbiedingen.",
  moreInfoLink: "https://coolblue.nl",
  showAlternativeBanner: false
});
