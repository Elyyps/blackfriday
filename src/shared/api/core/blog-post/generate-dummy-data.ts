import { BlogPost } from "./blog-post";

export const generateDummyBlogPost = (): BlogPost[] => [
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/2.png",
    title: "Vijf voorspellingen van Black Friday Nederland",
    content:
      "Vorige week spraken we over de beste topdeals van vorig jaar. Maar wat kunnen we dit jaar verwachten met Black Friday 2018... ",

    link: { title: "Lees meer", url: "/blog-single" }
  },
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/4.png",
    title: "Top Black Friday deals van vorig jaar",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    link: { title: "Lees meer", url: "/blog-single" }
  },
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/3.png",
    title: "Top Black Friday deals van vorig jaar",
    content:
      "Nog even aftellen en Black Friday 2018 gaat beginnen. Zit jij er al helemaal klaar voor? Om er zeker van te zijn dat je geen enkele deal mist...",

    link: { title: "Lees meer", url: "/blog-single" }
  },
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/6.png",
    title: "Mis geen enkele Black Friday deal!",
    content:
      "Maximaal profiteren van Black Friday 2018?! Hier vind je 5 goede redenen om nog even te wachten met geld uitgeven tot het Black Friday is... ",
    link: { title: "Lees meer", url: "/blog-single" }
  },

  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/5.jpg",
    title: "5 redenen om te wachten met shoppen tot Black Friday",
    content:
      "Zoveel mogelijk profiteren van Black Friday in 2018?! Hier vindt u 5 goede redenen om nog even te wachten met geld uitgeven tot het Black Friday is. De laatste maanden van het jaar vo.. ",

    link: { title: "Lees meer", url: "/blog-single" }
  }
];
export const generate2DummyBelgianBlogPosts = (): BlogPost[] => [
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/blog-single.png",
    title: "De beste Moederdag cadeau deals",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    link: { title: "Lees meer", url: "/blog-single" }
  },
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/1.png",
    title: "Black Friday wordt Black Week – deals beginnen eerder",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",

    link: { title: "Lees meer", url: "/blog-single" }
  }
];
export const generate2DummyDutchBlogPosts = (): BlogPost[] => [
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/dutch/4.png",
    title: "De beste Moederdag cadeau deals",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",
    link: { title: "Lees meer", url: "/blog-single" }
  },
  {
    image: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/dutch/2.jpg",
    title: "Black Friday wordt Black Week – deals beginnen eerder",
    content:
      "Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... ",

    link: { title: "Lees meer", url: "/blog-single" }
  }
];
