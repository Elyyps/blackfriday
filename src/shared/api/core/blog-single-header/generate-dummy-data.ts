import { BlogSingleHeader } from "./blog-single-header";
import Oasis from "@assets/oasis.jpg";

export const generateBlogSingleHeaderDummyData = (): BlogSingleHeader => ({
  title: "Black Friday 2019 blog",
  image: Oasis,
  links: [{ title: "Home", url: "" }, { title: "Alle blogs", url: "" }],
  button: { title: "", url: "" }
});
