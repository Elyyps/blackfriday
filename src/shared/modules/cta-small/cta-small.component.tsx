import * as React from "react";
import styles from "./cta-small-component.module.scss";
import { ShadowCardComponent } from "@app/core/shadow-card";
import { IconComponent } from "@app/core/icon/icon.component";
import { Button } from "@app/core/button";

export interface ICtaSmallComponentProps {
  buttonClick?: () => void;
  buttonTitle?: string;
  icon?: any;
  text?: string;
}

const CtaSmallComponent = (props: ICtaSmallComponentProps) => {
  const { buttonClick, buttonTitle, icon, text } = props;

  return (
    <div className={styles["cta-small"]}>
      <div className="uk-container">
        <ShadowCardComponent borderRadius={["2px"]}>
          <div className={styles["cta-small__wrapper"]}>
            {icon && (
              <div className={styles["icon"]}>
                <IconComponent icon={icon} size="32px" />
              </div>
            )}
            <div className={styles["cta-small__title"]}>{text}</div>
            {buttonTitle && <Button onClick={buttonClick} title={buttonTitle} variant="primary-brand" />}
          </div>
        </ShadowCardComponent>
      </div>
    </div>
  );
};

export { CtaSmallComponent };
