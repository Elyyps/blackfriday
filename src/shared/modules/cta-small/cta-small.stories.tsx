import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
// import { withProvider } from "@app/modules/core/storybook-provider";
import { CtaSmallComponent } from "./cta-small.component";
import IconDefault from "@assets/icons/sign.svg";

const buttonClick = () => (location.href = "#");
storiesOf("CtaSmall", module)
  .addDecorator(withA11y)
  // .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <CtaSmallComponent
      buttonClick={buttonClick}
      buttonTitle="Naar deals"
      text="Niet helemaal wat je zoekt? Geeft niks want we hebben meer!"
      icon={IconDefault}
    />
  ))
  .add("Emtpy implementation", () => <CtaSmallComponent />);
