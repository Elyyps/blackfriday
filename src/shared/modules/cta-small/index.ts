export * from "./cta-small.component";
