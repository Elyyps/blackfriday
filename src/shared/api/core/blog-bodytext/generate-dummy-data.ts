import { BlogBodyTextModule } from "./blog-bodytext";
import { generateSidebarData } from "../sidebar/generate-dummy-data";

export const generateDummyBlogBodyText = (): BlogBodyTextModule => ({
  bodyText: {
    html: `<h2>
        Hogere kortingen, meer producten en meer uiteenlopende productcategorieën. Dit gaat er gebeuren met Black
        Friday.
      </h2>
      <h3>1. Wees er snel bij, winkels beginnen eerder met Black Friday</h3>
      <p>
        Vorig jaar zagen we deze trend al opkomen. Winkels willen altijd graag de eerste zijn met hun deal, daarom
        beginnen een aantal winkels niet op Black Friday, maar vaak een paar dagen eerder. Het is daarom belangrijk dat
        je niet alleen op 23 november 2018 <a href="#">Black Friday Nederland</a> goed in de gaten houdt, maar ook
        in de week voor de kortingsdag van het jaar de website al iedere dag even checkt. Je weet nooit wanneer de deals
        online komen. Toch zien we wel dat de allerbeste deals pas echt vanaf Black Friday ingaan, maar wees er dan snel
        bij! Want voordat je het weet is jouw favoriete product uitverkocht.
      </p>
      <h3>2. Groots uitpakken met Telecom deals</h3>
      <p>
        Telecom deals zijn niet meer weg te denken van Black Friday. Wij verwachten dat er dit jaar nog meer Telecom
        deals online komen dan vorig jaar. Ben je dus op zoek naar een <a href="#">mobiel</a>? Wacht dan nog even
        tot Black Friday want we verwachten dat providers als <a href="#">Vodafone</a>, <a href="#">T-Mobile</a>
        , <a href="#">BEN</a> en <a href="#">Tele2</a> weer groots uit gaan pakken met de smartphone deals van
        2018. De Samsung Galaxy of de Apple Iphone zijn smartphones waar je altijd wel een fikse korting op kunt vinden.
        Kun je niet wachten? Bekijk dan alvast de <a href="#">Samsung Galaxy deals</a> van dit moment.
      </p>
      <h3>3. 45% meer deelnemende winkels in 2019</h3>
      <img src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/3.png" width="279px" height="146px" className="align-right" alt="image description" />
      <p>
        Naar verwachting doen er in 2018 bijna de helft meer winkels mee aan <a href="#">Black Friday 2018</a>.
        Daardoor zal het aantal deals ook bijna met de helft stijgen. Het is dus belangrijk dat je jouw favoriete winkel
        goed in de gaten houdt, want voordat je het weet wordt je ondergesneeuwd door de kortingen en kun je jouw deal
        niet meer vinden.
      </p>
      <h3>4. Meer lokale scherpe aanbiedingen</h3>
      <p>
        Behalve grote namen als <a href="#">adidas</a>, <a href="#">MediaMarkt</a>, <a href="#">Ziggo</a> en
        <a href="#">Bol.com</a> is het ook slim om dit jaar de kleine lokale winkels goed in de gaten te houden.
        Naar verwachting zullen de kleine winkels en boetiekjes in jouw dorp of stad ook meedoen aan Black Friday. Black
        Friday Nederland richt zich echter alleen op de landelijke bekende winkels, dus de aanbiedingen van dat kleine
        winkeltje in jouw dorp zul je zelf in de gaten moeten houden. Dit kan vaak al door de winkel te volgen op Social
        Media en even te vragen of ze dit jaar meedoen aan Black Friday.
      </p>
      <h3>5. Meer producten in reis-, interieur- en fashionbranche</h3>
      <p>
        Dit jaar verwachten wij dat vooral de reisbranche uit gaat pakken. Deze speler was er vorig jaar ook bij met
        onder andere merken als: <a href="#">Ryanair</a>, Hotels.com, <a href="#">Expedia</a> en Hilton. De
        verwachting is vooral dat er kortingen komen op vliegtickets en hotels met Black Friday 2018. Houd onze
        reispagina goed in de gaten voor een recent overzicht van alle <a href="#">reisaanbiedingen</a> met Black
        Friday.
      </p>`,
    bottomPadding: { mobilePadding: 32, desktopPadding: 72 },
    topPadding: { mobilePadding: 32, desktopPadding: 72 },
    containerAlignment: "center",
    containerSize: "full",
    bottomMargin: "",
    topMargin: "",
    id: "1",
    name: "BlogBodyText"
  },
  dateTime: "2 minuten leestijd",
  sidebarItems: generateSidebarData(),
  id: "1",
  name: "BlogBodyTextModule",
  bottomMargin: "",
  topMargin: ""
});

export const generateBlackFriday2019Text = (): BlogBodyTextModule => ({
  bodyText: {
    html: `<p><em>Profiteren van korting tijdens <strong>Black Friday 2019?!</strong> Hier lees je alles over de achtergronden van deze Amerikaanse traditie én hoe je er dit jaar zelf van profiteert.</em></p>
<p><strong><a href="#">Black Friday</a> is een Amerikaanse feestdag die altijd wordt gevierd op de dag na <a href="#">Thanksgiving</a>. De naam is ontstaan vanwege het feit dat het altijd heel erg druk was in de stad op de dag na Thanksgiving want iedereen was vrij. De dag na Thanksgiving was dus dé perfecte dag om koopjes te scoren en met scherpe deals voor de dag te komen. Nederland heeft deze traditie overgenomen en daardoor is het op de vierde vrijdag in november altijd gekkenhuis in de winkelstraten.</strong></p>
<p><img class="aligncenter wp-image-10314 size-full" src="https://www.blackfridaynederland.nl/wp-content/uploads/2019/05/Banner-black-friday-2019.png" alt="black friday 2019" width="750" height="250" srcset="https://www.blackfridaynederland.nl/wp-content/uploads/2019/05/Banner-black-friday-2019.png 750w, https://www.blackfridaynederland.nl/wp-content/uploads/2019/05/Banner-black-friday-2019-300x100.png 300w" sizes="(max-width: 750px) 100vw, 750px"></p>
<h2><strong>Black Friday in Nederland 2019</strong></h2>
<p>Black Friday 2019 valt dit jaar op <strong>29 november.</strong> Op deze dag stunten ook in Nederland weer een groot aantal (web)winkels met dé beste aanbiedingen. Bij de start van <a href="#">Black Friday Nederland</a> in 2015 waren er maar liefs 35 (web)winkels met Black Friday aanbiedingen in Nederland maar afgelopen jaar is dit aantal gestegen tot 166 deelnemers. We verwachten dat dit aantal in 2019 alleen maar meer toe gaat nemen.</p>
<h3><strong>Verwachtingen voor Black Friday 2019</strong></h3>
<p>Vorig jaar selecteerde Black Friday Nederland maar liefst meer dan 166 Nederlandse winkels met scherpe deals. Er waren veel goede aanbiedingen te vinden in onder andere de&nbsp;<a href="#">elektronica</a> en <a href="#">fashion</a>&nbsp;branche, maar ook de <a href="#">gaming</a>, <a href="#">speelgoed</a> en <a href="#">vrije tijd</a> branche hadden scherpe deals die je zeker niet wilde missen. Dit keer verwachten we dat Black Friday zeker weer veel mooie deals gaat brengen en het nog groter gaat worden dan het al is.</p>
<p>Denk bijvoorbeeld aan lokale, kleine ondernemers in je eigen winkelstraat die toch ook mee gaan doen aan de rage, of de top 5 aan (web)winkels die allemaal dé beste deal willen hebben. Het wordt dus heel belangrijk dat je scherp in de gaten houdt wat voor deal jij tijdens Black Friday wilt scoren want voordat je het weet is jouw koopje al&nbsp; verdwenen!</p>
<h3><strong>Grootste spelers tijdens Black Friday</strong></h3>
<p>Er zijn een aantal winkels bij wie je er nu al vanuit kunt gaan dat ze volgend jaar weer van de partij zijn. <a href="#">MediaMarkt</a> heeft bijvoorbeeld elk jaar met Black Friday scherpe elektronica aanbiedingen. Hetzelfde geldt&nbsp; voor <a href="#">Coolblue</a>. Huishoudelijke apparatuur zoals wasmachines, stofzuigers of <a href="#">elektronica</a> artikelen als laptops en televisies zijn ineens een stuk beter betaalbaar met Black Friday bij Coolblue. Verder is <a href="#">Bol.com</a> ook een grote speler op de retailermarkt waar je een paar weken voor Sinterklaas- en kerst goedkoop al je cadeautjes kunt bestellen. Behalve deze drie deelnemers zijn er natuurlijk nog honderden andere deelnemende winkels.</p>
<h3><strong>Profiteren van Black Friday 2019 – Alle deals op een rijtje</strong></h3>
<p>Om orde te creëren in de chaos van alle aanbiedingen tijdens Black Friday 2019, zetten wij de beste deals voor je in een handig overzicht. Dankzij onze contacten met alle (web)winkels zijn wij als eerste op de hoogte van de beste aanbiedingen tijdens Black Friday. Wij updaten regelmatig het winkeloverzicht met de meest scherpe deals zodat jij als eerste weet wat de beste aanbiedingen worden tijdens Black Friday. Check regelmatig het winkeloverzicht voor het meest recente overzicht zodat jij zeker weet dat er geen enkele deal aan je neus voorbijgaat. Abonneer je verder op de Deal Alerts. Door je te abonneren op deze nieuwsbrief ontvang je voorafgaand en tijdens Black Friday regelmatig alerts met de beste deals. Deze deals kun jij vervolgens helemaal zelf afstemmen op je interesses.</p>`,
    bottomPadding: { mobilePadding: 32, desktopPadding: 72 },
    topPadding: { mobilePadding: 32, desktopPadding: 72 },
    containerAlignment: "center",
    containerSize: "full",
    bottomMargin: "",
    topMargin: "",
    id: "1",
    name: "BlogBodyText"
  },
  dateTime: "",
  sidebarItems: generateSidebarData(),
  id: "1",
  name: "BlogBodyTextModule",
  bottomMargin: "48px",
  topMargin: ""
});
