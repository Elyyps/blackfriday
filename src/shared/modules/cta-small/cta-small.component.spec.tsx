import React from "react";
import { shallow } from "enzyme";
import { generateCtaSmallModule } from "@app/api/modules/cta-small/generate-dummy-data";
import { CtaSmallModuleComponent } from "./cta-small.component";

describe("[CtaSmallModuleComponent]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<CtaSmallModuleComponent ctaSmallModule={generateCtaSmallModule()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
