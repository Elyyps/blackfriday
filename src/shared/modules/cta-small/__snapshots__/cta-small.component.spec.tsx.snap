// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CtaSmallModuleComponent] should render component without crashing 1`] = `
<CtaSmallComponent
  background={
    Object {
      "backgroundColour": "#f7f7f7",
    }
  }
  bottomPadding={
    Object {
      "desktopSpacing": 72,
      "mobileSpacing": 48,
    }
  }
  buttonLink="/deals-overview"
  buttonTitle="Naar deals"
  icon="sign.svg"
  id="1"
  name="CtaSmallModule"
  text="Niet helemaal wat je zoekt? Geeft niks want we hebben meer!"
/>
`;
